crates/mun_compiler/src/diagnostics.rs

```diff
@@ -3,111 +3,510 @@ use mun_hir::{FileId, HirDatabase, HirDisplay, Module};
 use mun_syntax::{ast, AstNode, SyntaxKind};
 use std::cell::RefCell;
 
-mod emit;
+use annotate_snippets::snippet::{Annotation, AnnotationType, Slice, Snippet, SourceAnnotation};
 
-pub use emit::Emit;
-use mun_errors::{Diagnostic, Level};
+use unicode_segmentation::UnicodeSegmentation;
+
+/// Replace all tabs('\t') in string to spaces(' ')
+fn replace_tabs(text: &mut String) {
```

Comment:
It would be more Rusty to insert a String and return a String instead of mutating the `text` param.